server/game/cards/12-KotI/WhenIWoke.js

```diff
@@ -0,0 +1,40 @@
+const DrawCard = require('../../drawcard.js');
+
+class WhenIWoke extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                afterChallenge: event => event.challenge.challengeType === 'military' && event.challenge.loser !== undefined 
```

Comment:
Checking that a `loser` is truthy is a little more idiomatic for Javascript. It would also cover any case where `loser` was `null` instead of `undefined`

```suggestion
                afterChallenge: event => event.challenge.challengeType === 'military' && event.challenge.loser
```

or possibly:

```suggestion
                afterChallenge: event => event.challenge.challengeType === 'military' && !!event.challenge.loser
```